core/src/test/java/org/apache/iceberg/rest/TestRESTScanPlanning.java

```diff
@@ -0,0 +1,871 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.rest;
+
+import static org.apache.iceberg.catalog.CatalogTests.FILE_A;
+import static org.apache.iceberg.catalog.CatalogTests.FILE_A_DELETES;
+import static org.apache.iceberg.catalog.CatalogTests.FILE_A_EQUALITY_DELETES;
+import static org.apache.iceberg.catalog.CatalogTests.FILE_B;
+import static org.apache.iceberg.catalog.CatalogTests.FILE_B_DELETES;
+import static org.apache.iceberg.catalog.CatalogTests.FILE_B_EQUALITY_DELETES;
+import static org.apache.iceberg.catalog.CatalogTests.FILE_C;
+import static org.apache.iceberg.catalog.CatalogTests.FILE_C_EQUALITY_DELETES;
+import static org.apache.iceberg.catalog.CatalogTests.SCHEMA;
+import static org.apache.iceberg.catalog.CatalogTests.SPEC;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.assertj.core.api.Assumptions.assumeThat;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.DeleteFile;
+import org.apache.iceberg.FileScanTask;
+import org.apache.iceberg.MetadataTableType;
+import org.apache.iceberg.MetadataTableUtils;
+import org.apache.iceberg.Scan;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.TableScan;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.catalog.SessionCatalog;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.expressions.Expressions;
+import org.apache.iceberg.inmemory.InMemoryCatalog;
+import org.apache.iceberg.io.CloseableIterable;
+import org.apache.iceberg.io.CloseableIterator;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.rest.responses.ErrorResponse;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+import org.mockito.Mockito;
+
+public class TestRESTScanPlanning {
```

Comment:
Sorry if I missed this, do we have a test where files are produced under different partition specs?

Reply:
I didn't had it added :), good call out !